src/game/UICmdEdit.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+#include "GameDef.h"
+#include "N3Base/N3UIBase.h"
+#include "N3Base/N3UIString.h"
+#include "N3Base/N3UIEdit.h"
+
+//////////////////////////////////////////////////////////////////////
+
+class CUICmdEditDlg : public CN3UIBase {
+  public:
+    CN3UIString * m_pText_Title;
+    CN3UIButton * m_pBtn_Ok;
+    CN3UIButton * m_pBtn_Cancel;
+    CN3UIEdit *   m_pEdit_Box;
+    std::string   m_szArg1;
+
+  public:
+    void SetVisible(bool bVisible);
+    void Open(std::string msg);
```

Comment:
Since there's no need to copy or modify, this should be passed by const reference, i.e.:
```cpp
void Open(const std::string& msg);
```